test/test_torch.py

```diff
@@ -4385,37 +4385,40 @@ def test_narrow_empty(self, device):
             sz[d] = 0
             self.assertEqual(sz, y.size())
 
-    def test_index_copy(self, device):
-        num_copy, num_dest = 3, 20
-        dest = torch.randn(num_dest, 4, 5, device=device)
-        src = torch.randn(num_copy, 4, 5, device=device)
-        idx = torch.randperm(num_dest, device=device).narrow(0, 0, num_copy)
-        dest2 = dest.clone()
-        dest.index_copy_(0, idx, src)
-        for i in range(idx.size(0)):
-            dest2[idx[i]] = src[i]
-        self.assertEqual(dest, dest2, atol=0, rtol=0)
-
-        dest = torch.randn(num_dest, device=device)
-        src = torch.randn(num_copy, device=device)
-        idx = torch.randperm(num_dest, device=device).narrow(0, 0, num_copy)
-        dest2 = dest.clone()
-        dest.index_copy_(0, idx, src)
-        for i in range(idx.size(0)):
-            dest2[idx[i]] = src[i]
-        self.assertEqual(dest, dest2, atol=0, rtol=0)
-
-        # Bool tensor
-        dest = torch.zeros(2, 2, dtype=torch.bool, device=device)
-        src = torch.tensor([[True, True], [True, True]], device=device)
-        index = torch.tensor([0, 1], device=device)
-        dest.index_copy_(0, index, src)
-        self.assertEqual(dest, torch.tensor([[True, True], [True, True]], device=device))
-
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_index_copy(self, device, dtype):
```

Comment:
Is there a test case for correct 0-dim handling?

Reply:
These two test those cases:

https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L3328

Reply:
Please add those test cases to the sample inputs and remove the method_tests() entries.

They'll still need an explicit test here, though, because sample inputs only validate that autograd and the jit work correctly. They don't check that the function computes the correct value in those cases.

Reply:
Now it tests all the empty cases correctly. I deleted these two legacy tests.
You'll be pleased to know (or not really) that this PR is also affected by the weird issue #49707

Reply:
@Lezcano that issue has an easy fix https://github.com/pytorch/pytorch/issues/49707#issuecomment-788140578, but I don't know if it's better to make it in a separate PR or here.

Reply:
I would be happy to add the fix to this PR. What do you think @mruberry ?
Other possibility would be to land this PR and then have another PR with the fix proposed by @ngimel + uncomment the relevan tests for both `index_copy` and `index_select`.

Reply:
Let's have a fix in a follow-up PR then. Thanks @Lezcano !